libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -433,6 +437,54 @@ void AP_DDS_Client::on_topic (uxrSession* uxr_session, uxrObjectId object_id, ui
 
 }
 
+#include "AP_DDS_Service_Table.h"
+
+void AP_DDS_Client::on_request([[maybe_unused]] uxrSession* uxr_session, uxrObjectId object_id, [[maybe_unused]] uint16_t request_id, SampleIdentity* sample_id, ucdrBuffer* ub, [[maybe_unused]] uint16_t length, void* args){
+    switch (object_id.id){
+        case services[to_underlying(ServiceIndex::ARMING_MOTORS)].rep_id:
+        bool arm;
+        const bool deserialize_success = ucdr_deserialize_bool(ub,&arm);
+        if(deserialize_success == false){
+            break;
+        }
+        
+        if(arm == true) {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Request for arming recieved");
+        } else {
+            GCS_SEND_TEXT(MAV_SEVERITY_INFO,"DDS Client: Request for disarming recieved");
+        }
+        
+        const uxrObjectId replier_id = {
+            .id = services[to_underlying(ServiceIndex::ARMING_MOTORS)].rep_id,
+            .type = UXR_REPLIER_ID
+        };
+        uint8_t reply_buffer[8] = {
+            0
+        };
+        ucdrBuffer reply_ub;
+
+        // dummy code
+        // TODO : add arming/disarming checks here 
+        bool result = arm;
```

Comment:
you should be able to call to AP_Arming like this:
 - AP::arming().arm(AP_Arming::Method::DDS)
 - AP::arming().disarm(AP_Arming::Method::DDS)

you will need to add DDS as an arming method in AP_Arming.h

Reply:
Done.. Thanks for this @tridge !